src/axom/spin/internal/linear_bvh/QueryAccessor.hpp

```diff
@@ -68,6 +68,40 @@ class QueryAccessor
                                const FloatType* z );
 
   /// @}
+
+  /// \name Ray Query Access Methods
+  /// @{
+
+  /*!
+   * \brief Gets the ray defined by a source point and normal direction.
+   *
+   * \param [out] ray the ray where to pack the source point and normal.
+   * \param [in] idx the index of the ray w.r.t. the supplied arrays.
+   * \param [in] x0 array of the x-coordinate source point locations.
+   * \param [in] nx array of x-components of the ray normal directions.
+   * \param [in] y0 array of the y-coordinate source point locations.
+   * \param [in] ny array of y-components of the ray normal directions.
+   * \param [in] z0 array of of the z-coorindate source point locations.
+   * \param [in] nz array of z-components of the ray normal directions.
+   *
+   * \tparam RayType the data-structure to hold a ray.
+   *
+   * \note The RayType must overload the `[]` operator for accessing the
```

Comment:
Does ``primal::Ray`` do this? It doesn't look like you are modifying ``primal::Ray`` in this PR.

Reply:
I opted not to use `primal::Ray` in this PR since I have something that already works and is sufficient for my use.